src/spherical_mesh.cc

```diff
@@ -31,15 +31,19 @@ std::shared_ptr<Mesh> SphericalMesh::slice(int start, int end) const
 
 double SphericalMesh::area(int i) const
     {
-    const double r = center(i);
+    const double r = lower_bound(i);
     return 4.*M_PI*r*r;
     }
 
-
+double SphericalMesh::volume() const
+    {
+    return (4.*M_PI/3.)*(L_*L_*L_);
```

Comment:
This will fail if the domain is shifted (`origin_ != 0`). This should always work, I think:
```suggestion
    const double rlo = lower_bound(0);
    const double rhi = upper_bound(shape_-1);
    return (4.*M_PI/3.)*(rhi*rhi*rhi - rlo*rlo*rlo);
```